tests/regression/results.csv

```diff
@@ -236,17 +236,17 @@ silesia,                            level 1,                            advanced
 silesia,                            level 3,                            advanced one pass,                  4849553
 silesia,                            level 4,                            advanced one pass,                  4786968
 silesia,                            level 5 row 1,                      advanced one pass,                  4640752
-silesia,                            level 5 row 2,                      advanced one pass,                  4638961
```

Comment:
This PR introduces a lot of insignificant changes (which is fine). Can you point out exactly which lines introduce functional changes?

Even better would be to separate into two commits (or two PRs). One to switch to `ZSTD_ParamSwitch_e`, and one to make the functional change.

Reply:
Turns out I introduced a bug where I was incorrectly passing in the value of the `useRowMatchFinder` param into the function `resolveBlockSplitterMode`, causing block splitter to always get enabled when row hash is. Perils of copy-paste..

I split this PR into two commits, one for ldm, since it's a bit different, and one for the rest. Basically there are no real changes, and every change that isn't a direct name replacement is a small adjustment in order to accommodate the naming change or int->enum changes. All the `results.csv` changes are just switching between 1 and 2 in row match finder, since the meaning of 1 and 2 changed for rowhash's enum. Lmk if there's is a better scheme to divide this up for the sake of making review easier.